src/InsertionsClient/Api/Props/Models/PropsFileVariableReference.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Net.Insertions.Props.Models
+{
+    /// <summary>
+    /// Represents a reference to a variable defined in a props file
+    /// </summary>
+    public class PropsFileVariableReference
+    {
+        /// <summary>
+        /// Name of the defined variable
+        /// </summary>
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Desired value of the variable
+        /// </summary>
+        public string Value { get; private set; }
+
+        /// <summary>
+        /// The file that was referencing this variable. Used when resolving the correct scope
+        /// for the variable.
+        /// </summary>
+        public string ReferencedFilePath { get; private set; }
+
+
+        /// <summary>
+        /// Creates an instance of <see cref="PropsFileVariableReference"/>.
+        /// </summary>
+        /// <param name="name">Name of the variable</param>
+        /// <param name="value">Desired value of the variable</param>
+        /// <param name="referencedFilePath">The file that was referencing this variable.</param>
+        public PropsFileVariableReference(string name, string value, string referencedFilePath)
```

Comment:
Since this is a model, I think it should be the caller's responsibility to pass the correct values (empty string, invalid path etc).

These are not nullables (`string?`), so we shouldn't get `null` values here.